### ISSUE TITLE
Monitoring V2 Grafana storage causes "breaks non-root policy" error

### ISSUE THREAD

**Issue by benzman81** [rancher/rancher#29639]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Install rancher-monitoring with selected storage for grafana using `Statefulset template` and use fix for this bug #29638.

**Result:**
Volume for grafana will be created but grafana cannot start due to error `Containers with incomplete status: [init-chown-data grafana-sc-datasources]` and `Error: container's runAsUser breaks non-root policy`.

**Other details that may be helpful:**
We use nfs storage with nfs-client-provisioner.

If we add `initChownData.enabled` to grafana while installing rancher-monitoring it works.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.5.1/k8s 1.19.3
- Installation option (single install/HA):
Single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Nodes have version from 17.9.1, 18.9.0 and 19.3.5
Rancher server is docker version 17.03.2-ce
```

**Comment by jiaqiluo** [comment]
I hit the same error when installing Monitoring v2 in v2.5-61a8a3691063108bc775c8e4e6f6db971541c253-head

- install Longhorn
- install Monitoring v2 `9.4.201`
- When installing monitoring v2, enable the persistent storage for Grafana

Results:

```
Warning | Failed | Error: container's runAsUser breaks non-root policy
```

<img width="1378" alt="Screen Shot 2020-10-21 at 2 43 41 PM" src="https://user-images.githubusercontent.com/6218999/96791841-0e4b8680-13ae-11eb-8bff-01ba5ab44c54.png">

**Comment by simonwh** [comment]
I have the same issue, and when using StatefulSet, I'm not allowed to update the "Run as non-root":
`StatefulSet.apps "rancher-monitoring-grafana" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden`

**Comment by StrongMonkey** [comment]
@jiaqiluo Fix is merged, can you retest this? Also make sure the chart can pass CIS again as I modified some securityContext which might impact CIS.

**Comment by jiaqiluo** [comment]
The bug fix is validated in rancerh `v2.5-d87f58ac300192c90377faef694efb796a824520-head` with the latest monitoring v2 chart `9.4.201` 

Monitoring v2 is installed successfully in a hardened cluster and is functional. 
<img width="1373" alt="Screen Shot 2020-10-23 at 11 53 58 AM" src="https://user-images.githubusercontent.com/6218999/97042832-7a50fa80-1526-11eb-9fb2-624e3475efef.png">

related UI bug:  https://github.com/rancher/dashboard/issues/1689
